packages/harpparts/src/apparatus/utils/reed-array-to-matrices/reed-array-to-matrices.ts

```diff
@@ -0,0 +1,56 @@
+import { pivotReedArray } from '../pivot-reed-array'
+import { mapReedPairToHole } from '../map-reed-pair-to-hole'
+import { mapHoleToIncludeValvebends } from '../map-hole-to-include-valvebends'
+import { mapHoleToIncludeOverbends } from '../map-hole-to-include-overbends'
+import { mapHoleToIncludeBends } from '../map-hole-to-include-bends'
+import { mapHoleToFilterOverbends } from '../map-hole-to-filter-overbends'
+import { mapHoleToFilterIfValved } from '../map-hole-to-filter-if-valved'
+import { mapHoleTierToInteractionid } from '../map-hole-tier-to-interactionid'
+import { mapHoleTierToHalfstepindex } from '../map-hole-tier-to-halstepindex'
+import { getHoleArrayErrorMessages } from '../get-hole-array-error-messages'
+import { deriveMatrixSpecs } from '../derive-matrix-specs/derive-matrix-specs'
+import type { HoleArray } from '../../types'
+import type { Apparatus } from '../../types'
+import type { TuningIds, ReedArray } from '../../../tuning'
+
+export const reedArrayToMatrices = (
+  reedArray: ReedArray,
+  tuningId: TuningIds,
+  isHalfValved?: boolean
+): Pick<Apparatus, 'halfstepIndexMatrix' | 'interactionMatrix'> => {
+  const holeArray = pivotReedArray(reedArray)
+    .map(mapReedPairToHole)
+    .map(mapHoleToIncludeBends)
+    .map(mapHoleToIncludeOverbends)
+    .map(mapHoleToFilterOverbends)
+    .map((hole) => (isHalfValved ? mapHoleToIncludeValvebends(hole) : hole))
```

Comment:
Should the `mapHoleToIncludeValvebends` function actually receive the valving type and make the decision about whether to modify the `Hole` or not?

Reply:
Done